tests/test_frame.py

```diff
@@ -1271,6 +1271,27 @@ def test_read_excel() -> None:
         df17: dict[int | str, pd.DataFrame] = pd.read_excel("foo", sheet_name=None)
 
 
+def test_read_excel_io_types() -> None:
+    # GH 195
+    df = pd.DataFrame([[1, 2], [8, 9]], columns=["A", "B"])
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".xlsx") as file:
+        as_str: str = file.name
+        df.to_excel(file)
```

Comment:
Just insert `file.close()` here and then you don't need the `try`..`finally` block.   That's what worked elsewhere.

Reply:
I'm not sure that's right --

- If we pass delete=False to NamedTemporaryFile, then we have to take responsibility for the file or else it will get left behind. Exiting the context block just closes the file and does not delete it.

- If we pass delete=True (the default) to NamedTemporaryFile, calling file.close() will unlink the file -- NamedTemporaryFile returns a [_TemporaryFileWrapper](https://github.com/python/cpython/blob/d4792ce916b94d090b6c7bce8b0f973e840c9e4e/Lib/tempfile.py#L459) which overrides close(). In this case we can't close the file without unlinking it, so the tests will fail because the file will be gone.

So we need to pass delete=False so we can access the file after we close it, which means that we need to take some explicit action to clean up the file.

I think some other tests are calling delete=False and then not cleaning up the file, which I think is what you wanted to avoid here.

Reply:
> I think some other tests are calling delete=False and then not cleaning up the file, which I think is what you wanted to avoid here.

Yes, I agree, and I created #196 so we remember to address that. 

I'm trying to figure out a way to address it in a uniform way.   So we could find a way here in this PR, and then address the other places later.

If you removed `try` and `finally` and just had `Path(as_str).unlink()` at the end, would that work?  Or is the concern that if the test fails when we run `pytest`, the file would still be left behind?

Reply:
Roger!

> If you removed try and finally and just had Path(as_str).unlink() at the end, would that work? Or is the concern that if the test fails when we run pytest, the file would still be left behind?

Yes, that works fine unless an assertion fails.

Reply:
> > If you removed try and finally and just had Path(as_str).unlink() at the end, would that work? Or is the concern that if the test fails when we run pytest, the file would still be left behind?
> 
> Yes, that works fine unless an assertion fails.

I'm leaning towards leaving out `try` and `finally` because if it does fail, you can at least see the file if you want to debug the issue.